app/src/main/java/com/mithridat/nonoconverter/backend/nonogram/Nonogram.java

```diff
@@ -0,0 +1,296 @@
+     */
+    public int getTopColsLength() {
+        return _top.length;
+    }
+
+    /**
+     * Method for getting length of _left or _top
+     *
+     * @param type - ROW, if _left
+     *               COL, if _top
+     * @return length of _left or _top
+     */
+    public int getFirstLength(int type) {
+        if (type == Field.ROW) return getLeftRowsLength();
+        return getTopColsLength();
+    }
+
+    /**
+     * Method for getting number of columns
+     * in left nonogram grid row with number i
+     *
+     * @param i - row number
+     * @return number of columns
+     */
+    public int getLeftRowLength(int i) {
+        return _left[i].length;
+    }
+
+    /**
+     * Method for getting number of rows
+     * in top nonogram grid column with number i
+     *
+     * @param i - column number
+     * @return number of rows
+     */
+    public int getTopColLength(int i) {
+        return _top[i].length;
+    }
+
+    /**
+     * Method for getting length of _left[i] or _top[i]
+     *
+     * @param i - i
+     * @param type - ROW, if _left
+     *               COL, if _top
+     * @return length of _left[i] or _top[i]
+     */
+    public int getSecondLength(int i, int type) {
+        if (type == Field.ROW) return getLeftRowLength(i);
+        return getTopColLength(i);
+    }
+
+    /**
+     * Method for getting left nonogram grid cell
+     * with row number i and column number j
+     *
+     * @param i - row number
+     * @param j - column number
+     * @return left nonogram grid cell with row number i and column number j
+     */
+    public int getLeftRowValue(int i, int j) {
+        return _left[i][j];
+    }
+
+    /**
+     * Method for getting top nonogram grid cell
+     * with column number i and row number j
+     *
+     * @param i - column number
+     * @param j - row number
+     * @return top nonogram grid cell with column number i and row number j
+     */
+    public int getTopColValue(int i, int j) {
+        return _top[i][j];
+    }
+
+    /**
+     * Method for getting _left[i][j] or _top[i][j]
+     *
+     * @param i - i
+     * @param j - j
+     * @param type - ROW, if _left
+     *               COL, if _top
+     * @return _left[i][j] or _top[i][j]
+     */
+    public int getValue(int i, int j, int type) {
+        if (type == Field.ROW) return getLeftRowValue(i, j);
+        return getTopColValue(i, j);
+    }
+
+    /**
+     * Method for checking correctness of filling of the field
+     *
+     * @return true, if filling of the field is correct
+     *         false, otherwise
+     */
+    public boolean checkCorrectness() {
+        int rows = _field.getRows(), cols = _field.getCols();
+        for (int i = 0; i < rows; i++) {
+            for (int j = 0; j < cols; j++) {
+                if (_field.getColor(i, j) == Field.EMPTY) {
+                    _field.setColor(i, j, 0);
+                }
+            }
+        }
+        return isEqual(new Nonogram(_field));
```

Comment:
Is it possible to not create a new instance of the class `Nonogram` to check correctness?